lib/snapshot-manager.js

```diff
@@ -365,8 +386,31 @@ function determineSnapshotDir(projectDir, testDir) {
 	return testDir;
 }
 
+function determineSourceMappedDir(options) {
+	const sourceMapFilePath = path.join(options.testDir, `${options.name}.map`);
+	const sourceMapContent = tryReadSourceMap(sourceMapFilePath);
+
+	if (sourceMapContent) {
+		const targetFilename = path.basename(sourceMapContent.file, 'js');
+		const testFileSource = sourceMapContent.sources.find(file => {
+			const fileName = file.replace(path.extname(targetFilename), '');
+			return fileName.indexOf(targetFilename) > -1;
+		});
```

Comment:
This logic isn't quite right. According to the [source map "spec"](https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/view) `file` is optional. There might be a `sourceRoot` which should be prepended to the `sources`.

I couldn't quite find a library which exports the correct source resolution. I think this will be sufficient for our use case (though I haven't tested it):

```js
// Assume there is but a single source. Multiple sources imply multiple possible
// snapshot locations, which won't work.
const firstSource = sourceMapContent.sources[0];
const rootedSource = sourceMapContent.sourceRoot ?
  sourceMapContent.sourceRoot + firstSource :
  firstSource;
// `convert-source-map` doesn't provide the path to the source map file, if the
// source map came from such a file. Assuming that if it did, it is stored right
// next to the test file. Or, alternatively, if it is stored elsewhere,
// resolving `rootedSource` relative to the test file results in a path that
// is the same as if it had been resolved relative to the source map file.
// Note that `options.relFile` is relative to `options.projectDir`.
const sourceFile = path.resolve(path.join(options.projectDir, options.relFile), rootedSource);
return path.dirname(sourceFile);
```

I'm not sure how much we need to guard against malformed source maps (e.g. no or empty `sources`), or source maps that unexpectedly resolve to an HTTP source.